app/src/main/java/org/mozilla/fenix/compose/Chip.kt

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.capitalize
+import androidx.compose.ui.text.intl.Locale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Default layout for a clickable chip.
+ *
+ * @param text [String] displayed in this chip. Ideally should only be one word.
+ * @param modifider [Modifier] used to be applied to the layout of the chip.
+ * @param textColor optional text [Color] for the chip.
+ * @param backgroundColor optional background [Color] for the chip.
+ * @param onClick optional Callback for when the user taps this chip.
+ */
+@Composable
+fun Chip(
+    text: String,
+    modifier: Modifier = Modifier,
+    textColor: Color = FirefoxTheme.colors.textActionPrimary,
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    onClick: (() -> Unit)? = null,
+) {
+    Box(
+        modifier = modifier
+            .selectable(false) {
+                if (onClick != null) {
+                    onClick()
+                }
+            }
+            .clip(MaterialTheme.shapes.small)
+            .background(color = backgroundColor)
+            .padding(horizontal = 16.dp, vertical = 10.dp),
```

Comment:
Let's instead have the root Composable be the `androidx.compose.material.Chip`, and we'll simply wrap and extend the functionality around that.